DOC: Add text about licensing context (#5658)
Slicer/Slicer: Docs/user_guide/about.md

## License

The 3D Slicer software is distributed under a BSD-style open source license that is broadly compatible with the Open Source Definition by [The Open Source Initiative](http://opensource.org/) and contains no restrictions on legal uses of the software.

To use Slicer, please read the [3D Slicer Software License Agreement](https://github.com/Slicer/Slicer/blob/master/License.txt) before downloading any binary releases of the Slicer.

### Historical notes about the license

The Slicer License was drafted in 2005 by lawyers working for Brigham and Women's Hospital (BWH), a teaching affiliate of Harvard Medical School, to be BSD-like but with a few extra provisions related to medical software.  It is specific to BWH so it's not directly reusable, but it could serve as a template for projects with similar goals.

> It was written in part because BWH was the prime contractor on an NIH-funded development consortium ([NA-MIC](na-mic.org)) and wanted all code contributions to be compatible with ultimate use in real-world medical products (that is, commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software).  Compliance with the Slicer License was required for subcontractors, a group that included General Electric and several universities (MIT, UNC...) who all reviewed and accepted this license.

Reviewer: ```suggestion
It was written in part because BWH was the prime contractor on an NIH-funded development consortium ([NA-MIC](na-mic.org)) and wanted all code contributions to be compatible with ultimate use in real-world medical products (that is, commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software).  Compliance with the Slicer License was required for subcontractors, a group that included GE Research and several universities (MIT, UNC...) who all reviewed and accepted this license.
```

Suggested by @wschroed

The license has been in continuous use since 2005 for the 3D Slicer software package (slicer.org) that as of 2021 has been downloaded more than a million times and has been referenced in about 12,000 academic publications (https://www.slicer.org/wiki/Main_Page/SlicerCommunity). Some of the code is also now being used in several medical products for which this license has been reviewed and accepted by the companies involved.

### License terms and reasons

Here are some of the key points that BWH included to make the license suit the case of a large hospital distributing open source medical software.

> * Contributors affirm that they have de-identified any patient data they contribute to avoid issues with HIPAA or related regulations.

Reviewer 2: I like the whole set of changes in this pull request very much. It is very clear and accurate description of the license.

I would just recommend to separate in this section more clearly terms that affect users and terms that are only for contributors. We should start with users, as most people are just users.

Author: We were waiting for the OSI team to summarize their discussions to their wiki, but that appears to take longer than expected.  I would say let's finalize and merge this.  We can always update it later as needed.

Reviewer 2: Agreed.

To use Slicer, please read the [3D Slicer Software License Agreement](https://github.com/Slicer/Slicer/blob/master/License.txt) before downloading any binary releases of the Slicer.

> ### Licensing Context

Reviewer 2: "Licensing context" sounds like some additional licensing terms or conditions, which could make things sound more complicated. I would make the section title less official-sounding, something like "Why a custom open-source license?" or something that makes it clear that this is some additional background information ("Rationale", "Motivation", "Background", ...).

The Slicer License was drafted in 2005 by lawyers working for Brigham and Women's Hospital (BWH), a teaching affiliate of Harvard Medical School, to be BSD-like but with a few extra provisions related to medical software.  It is specific to BWH so it's not directly reusable, but it could serve as a template for projects with similar goals.

It was written in part because BWH was the prime contractor on an NIH-funded development consortium (na-mic.org) and wanted all code contributions to be compatible with ultimate use in real-world medical products (that is, commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software).  Compliance with the Slicer License was required for subcontractors, a group that included General Electric and several universities (MIT, UNC...) who all reviewed and accepted this license.

> The license has been in continuous use since 2005 on software (slicer.org) that has been downloaded something like a million times and has been referenced in about 12,000 academic publications (https://www.slicer.org/wiki/Main_Page/SlicerCommunity).  Some of the code is also now being used in several medical products, with this license having been reviewed and accepted by the companies involved.

Reviewer: ```suggestion
 The license has been in continuous use since 2005 on a software package (slicer.org) that has been downloaded more than a million times and has been referenced in about 12,000 academic publications (https://www.slicer.org/wiki/Main_Page/SlicerCommunity).  Some of the code is also now being used in several medical products in which this license has been reviewed and accepted by the companies involved.
```

Here are some of the key points that BWH included to increase their comfort at distributing open source medical software.

* contributors affirm that they have de-identified any patient data they contribute

* contributors explicitly grant royalty free rights if they contribute code covered by a patent they control (i.e. to avoid submarine patents)

> * no GPL or other copyleft allowed because that would make the code basically impossible to use in real medical products

Reviewer 3: What's the definition of "real medical products"? 

Does the term "basically impossible" mean that there are workarounds or is it actually "impossible"?

Author: "real medical products" and "basically impossible" reference the motivation above where it says "commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software"

I don't know of any GPL code being used in regulated medical devices.  The people I know who develop such products would not want to be obligated to share their code the way GPL requires so they are very careful to avoid any such code.  Code like the Linux kernel, which has [explicit exceptions](https://www.kernel.org/doc/html/latest/process/license-rules.html) to make things more flexible.

Reviewer 3: Ok it might be helpful to organize the key points as the "What" is different and then the "Why". 

The following includes one of the "why"s for BWH decision to create this license, but really the decision was because of the 4 bullet points, not just the one mentioned in this sentence. Maybe this one was the primary reason and should be listed as the first "key" point?

> It was written in part because BWH was the prime contractor on an NIH-funded development consortium (na-mic.org) and wanted all code contributions to be compatible with ultimate use in real-world medical products (that is, commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software).

Something like:
- no GPL or other copyleft allowed. 
  - This was in part because BWH was the prime contractor on an NIH-funded development consortium (na-mic.org) and wanted all code contributions to be compatible with ultimate use in real-world medical products (that is, commercial FDA-approved medical devices, which are almost universally closed source even if they build on open software)

Reviewer 2: Instead of "basically impossible" it would be more accurate to say that "the risk of legal issues when using GPL licensed code in a commercial product is too high, therefore most companies prohibit their software developers from using such code".

> * no GPL or other copyleft allowed because that would make the code basically impossible to use in real medical products

Reviewer: ```suggestion
* no GPL or other copyleft allowed because that would make the code basically impossible to use in medical products integrating private IP
```

The license has been in continuous use since 2005 for the 3D Slicer software package (slicer.org) that as of 2021 has been downloaded more than a million times and has been referenced in about 12,000 academic publications (https://www.slicer.org/wiki/Main_Page/SlicerCommunity). Some of the code is also now being used in several medical products for which this license has been reviewed and accepted by the companies involved.

### License terms and reasons

Here are some of the key points that BWH included to make the license suit the case of a large hospital distributing open source medical software.

* Contributors affirm that they have de-identified any patient data they contribute to avoid issues with HIPAA or related regulations.

* Contributors explicitly grant royalty free rights if they contribute code covered by a patent they control (i.e. to avoid submarine patents).

> * No GPL or other copyleft code is allowed because that could make complicated and risky to mix Slicer code with private intellectual property, which is often present in regulated medical products.

Reviewer 3: ```suggestion
* No GPL or other copyleft code is allowed because that could make it complicated and risky to mix Slicer code with private intellectual property, which is often present in regulated medical products.
```
